src/react/cache/QueryReference.ts

```
@@ -201,10 +201,7 @@ export class InternalQueryReference<TData = unknown> {
       return;
     }
 
-    this.result = result;
-    this.promise = result.data
-      ? createFulfilledPromise(result)
-      : createRejectedPromise(result);
```

Turns out we never reached this case before! We were throwing the entire result as the error rather than the `error` itself, which is a bug. This change in behavior exposed this issue.

---

If we never reached this before, why do we reach it now? 
If I'm not mistaken, you didn't change any code flow above this block - and there should be plenty of cases where `result.data` could be `undefined` (`getCurrentResult` should return `undefined` whenever the `data` would be `{}`).

---

Perhaps I'm making too much of an assumption with that claim. The reason I feel this way because this was throwing the `result` instead of the `error` and none of my tests for handling errors caught this. I have tests for all sorts of variations of errors (error on first render, error on refetch, error on incremental chunks of `@defer`, etc), so I think either the code block above this happened to catch all of those, or something else was amiss. I think the `result.data` just happened to be defined on all cases where it made it past line 195: https://github.com/apollographql/apollo-client/pull/11032/files/22b695c1bb80f0a6cde9b0be8f2573133174e081#diff-5aed35acafd928a4c69ca952cde66d25bd51dde880bff81a0801250841daf1b3L195

---

FYI, I'll be refactoring this logic anyways in likely my next PR, so expect this stuff to change and work more robustly. I had to get this change in there first to help that refactor.